test/e2e/extended_suite_test.go

```diff
@@ -7,18 +7,25 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/kelseyhightower/envconfig"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 var (
-	gitCommit   = "unknown"
-	kubeconfig  = flag.String("kubeconfig", "../../_data/_out/admin.kubeconfig", "Location of the kubeconfig")
-	artifactDir = flag.String("artifact-dir", "", "Directory to place artifacts when a test fails")
+	gitCommit  = "unknown"
+	testConfig Config
 )
 
 var _ = BeforeSuite(func() {
-	c = newTestClient(*kubeconfig, *artifactDir)
+	err := envconfig.Process("", &testConfig)
+	if err != nil {
+		panic(err)
+	}
+	if testConfig.ArtifactDir == "" {
+		testConfig.ArtifactDir = *flag.String("artifact-dir", "", "Directory to place artifacts when a test fails")
```

Comment:
You are missing `flag.Parse` but I am wondering what's the added value of maintaining two different ways of doing the same thing. I am not in love with environment variables because we need to be documenting them. Maybe revert the move of artifact-dir to an env since the issue was only about KUBECONFIG? I don't feel strong but I definitely don't want two separate ways to do the same thing.